File: app/src/main/java/tr/name/fatihdogan/books/api/response/DownloadAccess.java
package tr.name.fatihdogan.books.api.response;

import android.support.annotation.Keep;

@Keep
@SuppressWarnings({"unused", "WeakerAccess"})
public class DownloadAccess {

    public static final String KIND_BOOKS_DOWNLOAD_ACCESS_RESTRICTION = "books#downloadAccessRestriction";

    /**
     * {@link #KIND_BOOKS_DOWNLOAD_ACCESS_RESTRICTION}
     */
    public String kind;
    public String volumeId;
    public Boolean restricted;
    public Boolean deviceAllowed;
    public Boolean justAcquired;
    public Integer maxDownloadDevices;
    public Integer downloadsAcquired;
    public String nonce;
    public String source;
    public String reasonCode;
    public String message;
    public String signature;

}

File: app/src/main/java/com/fthdgn/books/api/response/DownloadAccess.kt
@file:Suppress("unused")

package com.fthdgn.books.api.response

import android.support.annotation.Keep

@Keep
class DownloadAccess {

    /**
     * [.KIND_BOOKS_DOWNLOAD_ACCESS_RESTRICTION]
     */
    var kind: String? = null
    var volumeId: String? = null
    var restricted: Boolean? = null
    var deviceAllowed: Boolean? = null
    var justAcquired: Boolean? = null
    var maxDownloadDevices: Int? = null
    var downloadsAcquired: Int? = null
    var nonce: String? = null
    var source: String? = null
    var reasonCode: String? = null
    var message: String? = null
    var signature: String? = null

    companion object {

        val KIND_BOOKS_DOWNLOAD_ACCESS_RESTRICTION = "books#downloadAccessRestriction"
    }

}
